spec/reek/source/source_locator_spec.rb

```diff
@@ -34,27 +34,57 @@
         test_configuration_for(CONFIG_PATH.join('with_excluded_paths.reek'))
       end
 
-      let(:path) { SAMPLES_PATH.join('source_with_exclude_paths') }
-
-      let(:expected_paths) do
-        [path.join('nested/uncommunicative_parameter_name.rb')]
-      end
-
-      let(:paths_that_are_expected_to_be_ignored) do
-        [
-          path.join('ignore_me/uncommunicative_method_name.rb'),
-          path.join('nested/ignore_me_as_well/irresponsible_module.rb')
-        ]
-      end
-
-      it 'does not use excluded paths' do
-        sources = described_class.new([path], configuration: configuration).sources
-        expect(sources).not_to include(*paths_that_are_expected_to_be_ignored)
-      end
-
-      it 'scans directories that are not excluded' do
-        sources = described_class.new([path], configuration: configuration).sources
-        expect(sources).to eq expected_paths
+      let(:options) { instance_double('Reek::CLI::Options', force_exclusion?: false) }
+
+      context 'when path is a file name' do
+        let(:path) { SAMPLES_PATH.join('source_with_exclude_paths','ignore_me','uncommunicative_method_name.rb') }
+
+        context 'when options.force_exclusion? is true' do
+          before do
+            allow(options).to receive(:force_exclusion?).and_return(true)
+          end
+
+          it 'excludes files in ignored directories' do
```

Comment:
I found this to be quite confusing. Above it says "when path is a file name". Ok, sounds good. Then it says down here "excludes files in ignored directories" and I'm thinking "wait, what? How can a file name exclude files in directories"?

How about 

>> it 'excludes this file'

to make this clearer?